cmd/juju/cloud/show.go

```diff
@@ -98,6 +127,32 @@ func (c *showCloudCommand) Run(ctxt *cmd.Context) error {
 	return nil
 }
 
+func (c *showCloudCommand) getCloud() (*cloudDetails, error) {
```

Comment:
This is better as 2 funcs - getLocalCloud() and getControllerCloud()
Do the controller == "" check above in Run()